src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
+
+    return response;
```

Comment:
PullProofAI

Consider using TypeScript interfaces instead of type aliases for `ControllerResponse` and `APIResponse`.
```typescript
export interface ControllerResponse {
  isSuccess: number;
  statusCode: number;
  message: string;
  data?: any;
  error?: any;
}

export const APIResponse = {
  create: (
    isSuccess: number,
    statusCode: number,
    message?: string,
    data?: any,
    error?: any,
  ): ControllerResponse => {
    const response: ControllerResponse = { isSuccess, statusCode, message };
    if (error) {
      response.error = error;
    } else if (!error && data) {
      response.data = data;
    } else {
      response.data = {};
    }

    return response;
  },
};
```
Using interfaces provides better clarity and allows for easier extension and type checking.


<!-- This is an auto-generated comment by PullProofAI -->